src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -43,51 +45,143 @@ def get(self, request, organization):
 class KeyTransactionEndpoint(KeyTransactionBase):
     permission_classes = (KeyTransactionPermission,)
 
+    def get(self, request, organization):
+        if not self.has_team_feature(request, organization):
+            return Response(status=404)
+
+        transaction_name = request.GET.get("transaction")
+        if transaction_name is None:
+            raise ParseError(detail="A transaction name is required")
+
+        project = self.get_project(request, organization)
+
+        key_teams = TeamKeyTransaction.objects.filter(
+            organization=organization,
+            team__in=request.access.teams,
+            project=project,
+            transaction=transaction_name,
+        )
+
+        return Response(serialize(list(key_teams)), status=200)
+
     def post(self, request, organization):
         """ Create a Key Transaction """
         if not self.has_feature(request, organization):
             return Response(status=404)
 
         project = self.get_project(request, organization)
 
-        base_filter = {"organization": organization, "owner": request.user}
+        if not self.has_team_feature(request, organization):
+            base_filter = {"organization": organization, "owner": request.user}
+
+            with transaction.atomic():
+                serializer = serializers.KeyTransactionSerializer(
+                    data=request.data, context=base_filter
+                )
+                if serializer.is_valid():
+                    data = serializer.validated_data
+                    base_filter["transaction"] = data["transaction"]
+                    base_filter["project"] = project
+
+                    if KeyTransaction.objects.filter(**base_filter).exists():
+                        return Response(status=204)
+
+                    try:
+                        KeyTransaction.objects.create(**base_filter)
```

Comment:
What about using a KeyTransaction.objects.get_or_create pattern here?

Reply:
This is the existing `KeyTransaction`, going to leave that unchanged (at least in this PR).